packages/web3-eth-accounts/src/index.js

```diff
@@ -334,7 +333,7 @@ Accounts.prototype.recover = function recover(message, signature, preFixed) {
     var args = [].slice.apply(arguments);
 
 
-    if (_.isObject(message)) {
+    if (typeof message === 'object' && !!message) {
```

Comment:
I don't understand why `!!message` is necessary, an empty object is truthy (applies to all other occurrences of `!!`)